core/src/main/java/org/apache/accumulo/core/gc/ReferenceDirectory.java

```diff
@@ -46,8 +47,8 @@ public String getTabletDir() {
    * A Tablet directory should have a metadata entry equal to the dirName.
    */
   @Override
-  public String getMetadataEntry() {
-    if (!tabletDir.equals(metadataEntry)) {
+  public TabletFileMetadataEntry getMetadataEntry() {
+    if (!tabletDir.equals(metadataEntry.getFilePathString())) {
```

Comment:
Since the variables are `final`, I think this check should be done in the constructor.